state/application.go

```diff
@@ -465,25 +447,24 @@ func removeResourcesOps(st *State, applicationID string) ([]txn.Op, error) {
 // When force is set, the operation will proceed regardless of the errors,
 // and if any errors are encountered, all possible accumulated operations
 // as well as all encountered errors will be returned.
-// When 'force' is set, this call will return both operations to remove this
-// application as well as all operational errors encountered.
+// When 'force' is set, this call will return operations to remove this
+// application and will accumulate all operational errors encountered in the operation.
 // If the 'force' is not set, any error will be fatal and no operations will be returned.
-func (a *Application) removeOps(asserts bson.D, force bool) ([]txn.Op, []error, error) {
+func (a *Application) removeOps(asserts bson.D, op *ForcedOperation) ([]txn.Op, error) {
 	ops := []txn.Op{{
 		C:      applicationsC,
 		Id:     a.doc.DocID,
 		Assert: asserts,
 		Remove: true,
 	}}
 
-	errs := []error{}
 	// Remove application offers.
 	removeOfferOps, err := removeApplicationOffersOps(a.st, a.doc.Name)
 	if err != nil {
-		if !force {
-			return nil, errs, errors.Trace(err)
+		if !op.Force {
```

Comment:
A further refinement (which you don't need to do now) could be adding an IsFatal method to ForcedOperation - then this could be:
```go
if op.IsFatal(err) {
    return nil, errors.Trace(err)
}
```
And IsFatal would be:
```go
func (fo *ForcedOperation) IsFatal(err error) bool {
    if err == nil {
        return false
    }
    if fo.Force {
        return true
    }
    fo.AddError(err)
    return false
}
```

Reply:
Love it! I'll change it here in this PR

Reply:
Actually, I might land this as is and then do it on one of the follow ups.. This has been open too long for my liking \o/